### PR TITLE
grep.py 0.8

### PR DESCRIPTION
**Pull request by mickael9**
- use weechat 1.5+ api for background processing (old method was unsafe and buggy)
- add timeout_secs setting (was previously hardcoded to 5 mins)

### CODE REVIEW

File: python/grep.py [hunk 1]
```diff
@@ -66,6 +69,11 @@
 #
 #   History:
 #
+#   2017-09-20, mickael9
+#   version 0.7.9:
+#   * use weechat 1.5+ api for background processing (old method was unsafe and buggy)
```
Comment (Zarthus) [inline python/grep.py]: Does this change the minimum weechat script version possible? Version bump should at least be minor (or perhaps even major, due to bc break) - definitely not patch.

File: python/grep.py [hunk 2]
```diff
@@ -198,7 +206,7 @@
 ###
 
 from os import path
-import sys, getopt, time, os, re, tempfile
+import sys, getopt, time, os, re, cPickle
```
Comment (Zarthus) [inline python/grep.py]: cPickle seems to be a python 2 thing. In python 3 it's called pickle.

Perhaps you should do something like 

```py
pycompat = sys.version_info[0] < 3
if pycompat:
  import cPickle as pickle
else:
  import pickle

# and adapt the code to work in both py2 and py3 where API changes are necessary.
```

Comment (Zarthus) [reply python/grep.py]: Without this compatibility it is a breaking change as I think grep works on python 3. If it doesn't, movement towards 3.x support is still something good to be working towards.

File: python/grep.py (hunk 1 above)
Comment (mickael9) [reply python/grep.py]: I bumped it to 0.8, but I think users should be discouraged from using weechat <1.9.1 with all the security vulnerabilities

File: python/grep.py (hunk 2 above)
Comment (mickael9) [reply python/grep.py]: I added a fallback to `pickle` in case importing `cPickle` fails

**Pull request closed without merging**